tools/c7n_azure/c7n_azure/policy.py

```diff
@@ -270,13 +272,17 @@ def run_for_event(policy, event=None):
             return
 
         with policy.ctx:
+            rt = time.time() - s
+
             policy.ctx.metrics.put_metric(
                 'ResourceCount', len(resources), 'Count', Scope="Policy",
                 buffer=False)
-
+            policy.ctx.metrics.put_metric(
```

Comment:
I think we should upload metrics even if no resources were found (like 272 has early exit)

Reply:
Yeah pull mode does this, it breaks from lambda implementation further I think but that is fine.  Fixing.